frontend/src/libraries/wamp/types.ts

```diff
@@ -615,6 +496,18 @@ export type ProcedureTypes = {
 export type ProcedureType = keyof ProcedureTypes;
 
 export type ProcedureArgs<P extends ProcedureType> = ProcedureTypes[P]["args"];
+// WAMP cannot squeeze undefined through the communication channel
+// Though this check doesn't work at the moment
+// TODO: Investigate how to force typescript into this kind of checks
+// Minimal repro:
+// type NeverUndefined<T> = T extends undefined ? never : T;
+// type MyType = NeverUndefined<number | undefined> // should be never, actually number
 export type ProcedureResult<
   P extends ProcedureType
-> = ProcedureTypes[P]["result"];
+> = ProcedureTypes[P]["result"] extends undefined
+  ? never
+  : ProcedureTypes[P]["result"];
+
+// See https://stackoverflow.com/a/49402091/2017859
+export type KeysOfUnion<T> = T extends T ? keyof T : never;
+export * as RPC from "../../../../backend/src/rpc-types";
```

Comment:
It is quite concerning that we directly depend on backend source code in the frontend. At the very least, I would like to suggest to have a common package in the root of the project (ideally, added to both backend/frontend as a regular package [package.json])